src/lightning/pytorch/callbacks/model_checkpoint.py

```diff
@@ -604,6 +604,7 @@ def _find_last_checkpoints(self, trainer: "pl.Trainer") -> Set[str]:
                 os.path.normpath(p)
                 for p in self._fs.ls(ckpt_path, detail=False)
                 if self.CHECKPOINT_NAME_LAST in os.path.split(p)[1]
+                and os.path.split(p)[1].endswith(self.FILE_EXTENSION)
```

Comment:
```suggestion
                if f"{self.CHECKPOINT_NAME_LAST}{self.FILE_EXTENSION}" == os.path.split(p)[1]
```

Reply:
I am not sure if the assumption that the file is always called "last.chk" is correct or not. I thought that it could have other names as well like the epoch could be in the name, etc.

Reply:
@yassersouri are you still interested in finishing this PR?